issue_templates: Add bug and feature request templates. (#2056)
firecracker-microvm/firecracker: .github/ISSUE_TEMPLATE/feature_request.md

---
name: Feature request
about: Suggest an idea for this project

> title: "[Feature Request]"

Reviewer: ```suggestion
title: "[Feature Request] Title"
```

Author: I was under the impression that this text is just appended to the front of the text that people enter.  I wonder if your suggestion would give "[Feature Request] Title Please add some new feature"?

Author: Sorry,  think I misunderstood :-).  Done.

labels: ''
assignees: ''

---

> ## Is your feature request related to an existing problem?  Please describe.

Reviewer 2: I would open with a "Why is this important/What are the usecases"

Author: Amended

> `[Author TODO: A clear and concise description of what the problem is.  For example, I'm always frustrated when ...]`

Reviewer: ```suggestion
`[Author TODO: A clear and concise description of what the problem is.]`
```

Author: Done.

> ## Describe the solution you'd like ##

Reviewer: ```suggestion
## Describe the desired solution ##
```

Author: Done.

> `[Author TODO: A clear and concise description of what you want to happen.]`

Reviewer: ```suggestion
`[Author TODO: A clear and concise description of how you would like the feature to work.]`
```

Author: Done.

> ## Describe possible alternatives

Reviewer 2: Sometimes a feature would simplify the stack like for example macvtap vs tap. It would make sense that this paragraph refers also to how one works around this gap in functionality.

Author: I *think* I got your point,  but happy to change if not...

> ## Describe alternatives you've considered ##

Reviewer: ```suggestion
## Describe possible alternatives ##
```

Author: Done.

> `[Author TODO: A clear and concise description of any alternative solutions or features you've considered.]`

Reviewer: ```suggestion
`[Author TODO: A clear and concise description of any alternative solutions or features you have considered.]`
```

Author: Done

## Additional context

`[Author TODO: Add additional context about this feature request here.]`

## Checks
- [ ] Have you searched the Firecracker Issues database for similar requests?

> - [ ] Have you read all the existing relevant Firecracker documentation?

Reviewer 3: I would add a line which confirms the feature request is aligned with out [core tentes](https://github.com/firecracker-microvm/firecracker/blob/bef292499e2ad15ffd3229f7369564605a336b2c/CHARTER.md#tenets-unless-you-know-better-ones).

Author: Added.
